clippy_lints/src/matches.rs

```diff
@@ -712,6 +821,68 @@ fn check_wild_in_or_pats(cx: &LateContext<'_, '_>, arms: &[Arm<'_>]) {
     }
 }
 
+fn check_match_single_binding(cx: &LateContext<'_, '_>, ex: &Expr<'_>, arms: &[Arm<'_>], expr: &Expr<'_>) {
+    if in_macro(expr.span) || arms.len() != 1 || is_refutable(cx, arms[0].pat) {
+        return;
+    }
+    let matched_vars = ex.span;
+    let bind_names = arms[0].pat.span;
+    let match_body = remove_blocks(&arms[0].body);
+    let mut snippet_body = if match_body.span.from_expansion() {
+        Sugg::hir_with_macro_callsite(cx, match_body, "..").to_string()
+    } else {
+        snippet_block(cx, match_body.span, "..").to_owned().to_string()
+    };
+
+    // Do we need to add ';' to suggestion ?
+    match match_body.kind {
+        ExprKind::Block(block, _) => {
+            // macro + expr_ty(body) == ()
+            if block.span.from_expansion() && cx.tables.expr_ty(&arms[0].body).is_unit() {
```

Comment:
NIT: 
```suggestion
            if block.span.from_expansion() && cx.tables.expr_ty(&match_body).is_unit() {
```

Same below. Or is there a reason, that this isn't possible? Otherwise, I'd prefer using only `match_body` for this, to not depend on 2 different expressions.

Reply:
No problem at all. Using `match_body` makes me feel not using the original form of body, as it has been generated using `remove_blocks` function.